python/lsst/meas/astrom/astrometry_net.cc

```diff
@@ -0,0 +1,598 @@
+        solver_set_keep_logodds(_solver.get(), t);
+    }
+
+    void setPixelScaleRange(double lo, double hi) {
+        _solver->funits_lower = lo;
+        _solver->funits_upper = hi;
+    }
+
+    void setRaDecRadius(double ra, double dec, double rad) {
+        solver_set_radec(_solver.get(), ra, dec, rad);
+    }
+
+    void setImageSize(int W, int H) {
+        solver_set_field_bounds(_solver.get(), 0, W, 0, H);
+        double hi = hypot(W,H);
+        double lo = 0.1 * std::min(W,H);
+        solver_set_quad_size_range(_solver.get(), lo, hi);
+    }
+
+    void setMaxStars(int N) {
+        _solver->endobj = N;
+    }
+
+    void setStars(lsst::afw::table::SourceCatalog const & srcs, int x0, int y0) {
+        // convert to Astrometry.net "starxy_t"
+        starxy_free(_solver->fieldxy);
+        const size_t N = srcs.size();
+        starxy_t *starxy = starxy_new(N, true, false);
+        for (size_t i=0; i<N; ++i) {
+            double const x    = srcs[i].getX();
+            double const y    = srcs[i].getY();
+            double const flux = srcs[i].getPsfFlux();
+            starxy_set(starxy, i, x - x0, y - y0);
+            starxy_set_flux(starxy, i, flux);
+        }
+        // Sort the array
+        starxy_sort_by_flux(starxy);
+
+        starxy_free(solver_get_field(_solver.get()));
+        solver_free_field(_solver.get());
+        solver_set_field(_solver.get(), starxy);
+        solver_reset_field_size(_solver.get());
+        // Find field boundaries and precompute kdtree
+        solver_preprocess_field(_solver.get());
+    }
+
+private:
+    std::unique_ptr<solver_t, solver_deleter> _solver;
+};
+
+/*
+ * Wrap index_t
+ *
+ * This class has no constructor; the wrapper is only used for objects returned by MultiIndex::operator[]
+ */
+void declareIndex(py::module & mod) {
+    py::class_<index_t> cls(mod, "index_t");
+
+    cls.def("overlapsScaleRange", [](index_t & self, double qlo, double qhi) {
+        return index_overlaps_scale_range(&self, qlo, qhi);
+    }, "qlow"_a, "qhigh"_a);
+    cls.def("reload", [](index_t & self) {
+        if(index_reload(&self)) {
+            std::ostringstream os;
+            os << "Failed to reload multi-index file " << self.indexname;
+            throw LSST_EXCEPT(lsst::pex::exceptions::RuntimeError, os.str());
+        }
+    });
+
+    cls.def_readonly("indexid", &index_t::indexid);
+    cls.def_readonly("healpix", &index_t::healpix);
+    cls.def_readonly("hpnside", &index_t::hpnside);
+    cls.def_readonly("nstars", &index_t::nstars);
+    cls.def_readonly("nquads", &index_t::nquads);
+}
+
+/**
+ * Wrap MultiIndex, a thin shim around multiindex_t
+ */
+void declareMultiIndex(py::module & mod) {
+    py::class_<MultiIndex> cls(mod, "MultiIndex");
+
+    cls.def(py::init<std::string const &>(), "filepath"_a);
+
+    cls.def("__getitem__", &MultiIndex::operator[], "i"_a, py::return_value_policy::reference_internal,
+            py::is_operator());
```

Comment:
This could easily be done in Python, but I am not sure how to do it in C++. The existing Python code doesn't use it, so I'm inclined to skip it, since the current code works and we are trying get rid of astrometry_net.